pilot/pkg/networking/core/v1alpha3/configgen.go

```diff
@@ -15,8 +15,28 @@
 package v1alpha3
 
 import (
+	"github.com/golang/protobuf/ptypes/any"
+
 	meshconfig "istio.io/api/mesh/v1alpha1"
+	"istio.io/istio/pilot/pkg/model"
 	"istio.io/istio/pilot/pkg/networking/plugin"
+	"istio.io/istio/pilot/pkg/networking/util"
+)
+
+// To avoid a recoursive depenency to v2.
+const (
+	typePrefix = "type.googleapis.com/envoy.api.v2."
+
+	// Constants used for XDS
+
+	// ClusterType is used for cluster discovery. Typically first request received
+	ClusterType = typePrefix + "Cluster"
```

Comment:
can use go-control-plane `resource` package as well

Reply:
The go-control-plane is creating a big mess - with their use of golang protos when everything else is using gogo. 

I'm looking to reduce the deps, and switch to generated stubs using gogo at some point.